### PR TITLE
feat: support `snyk monitor --reachable-vulns`

### PR DESCRIPTION
**Pull request by darmalovan**
- [x] Ready for review
- [ ] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Support `snyk monitor --reachable-vulns` for maven behind a feature flag.

This change will swill send an additional payload in the form of
call graph to our backend.

#### What are the relevant tickets?
https://snyksec.atlassian.net/browse/FLOW-207

### CODE REVIEW

File: src/cli/commands/monitor/index.ts [hunk 1]
```diff
@@ -94,6 +95,11 @@ async function monitor(...args0: MethodArgs): Promise<any> {
         packageManager = detect.detectPackageManager(path, options);
       }
 
+      await validateOptions(
+        options as Options & MonitorOptions,
```
Comment (anthogez) [inline src/cli/commands/monitor/index.ts:99]: I think would be nice if you created a extended type in `src/cli/commands/monitor/types.ts` where you apply the `type intersection` between `Options & MonitorOptions` something like `GlobalOptions` or `ExtendedOptions`

Comment (darmalovan) [reply src/cli/commands/monitor/index.ts:99]: I think `GlobalOptions` might not be so useful from a user perspective, as this is the method used to show that it's only `Options & MonitorOptions` You can also have `Options & TestOptions`
This convention is used across the codebase - I don't think I want to change this as part of this PR.

File: src/lib/options-validator.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { MonitorOptions, Options, TestOptions } from './types';
+import * as config from './config';
+import { SupportedPackageManagers } from './package-managers';
+import * as reachableVulns from './reachable-vulns';
+
+export async function validateOptions(
+  options: (Options & TestOptions) | (Options & MonitorOptions),
```
Comment (anthogez) [inline src/lib/options-validator.ts:7]: Same here would be nice to have a type something like GlobalOptions?
`export type GlobalOptions = Options & MonitorOptions | Options & TestOptions;` ?
What do you think?

File: src/cli/commands/monitor/index.ts (hunk 1 above)
Comment (anthogez) [reply src/cli/commands/monitor/index.ts:99]: I understood your point, it's ok

**Pull request merged**